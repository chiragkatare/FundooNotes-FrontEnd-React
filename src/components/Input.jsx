import React, { Component } from "react";
import { TextField } from "@material-ui/core";


export default class Input extends Component {
    constructor(props) {
        super(props);
        this.state = {
            data: '',
            color:this.props.color
        }
        this.sendDataToParent = this.sendDataToParent.bind(this);
    }
    sendDataToParent(event) {
        
        this.setState({ data: event.target.value })
        this.props.onChange(event)
        // console.log("from input", this.state.data);
    }
    render() {
        return (
            <TextField 
                className='label'
                name={this.props.name}
                type={this.props.type}
                placeholder={this.props.placeholder}
                label={this.props.label}
                onChange={this.sendDataToParent}
                fullWidth
                color={this.state.color}
                required={this.props.required}
                // variant='outlined'
            />

        );
    }
}