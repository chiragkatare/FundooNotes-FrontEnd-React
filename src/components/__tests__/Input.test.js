import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount ,configure } from 'enzyme';
import Input from '../Input';

const mockFn = jest.fn();


// configure({ adapter: new Adapter() });




let props = {
    name:'password',
    type:'text',
    placeholder:'hello',
    label:'label',
    onChange:mockFn,
    
}

const component = mount(<Input {...props} />);

describe('Input', () => {
    it('should render correctly in "debug" mode', () => {

        // const component = mount(<Login debug />);
        expect(component).toMatchSnapshot();
      });

      it('should render correctly in with props', () => {
        
  
        let textfield = component.find('input');
        textfield.simulate('change',{target:{
            value:8,
        }})
        expect(component.state('data')).toEqual(8);
      });

      it('should should run the onchange props function', () => {
        
   
        let textfield = component.find('input');
        textfield.simulate('change',{target:{
            value:8,
        }})
        expect(mockFn.mock.calls.length).toEqual(2);
      });



});