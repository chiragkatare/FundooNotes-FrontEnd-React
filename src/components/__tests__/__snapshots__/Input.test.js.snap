// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly in "debug" mode 1`] = `
<Input
  label="label"
  name="password"
  onChange={[MockFunction]}
  placeholder="hello"
  type="text"
>
  <TextField
    className="label"
    fullWidth={true}
    label="label"
    name="password"
    onChange={[Function]}
    placeholder="hello"
    required={false}
    select={false}
    type="text"
    variant="standard"
  >
    <WithStyles(FormControl)
      className="label"
      fullWidth={true}
      required={false}
      variant="standard"
    >
      <FormControl
        className="label"
        classes={
          Object {
            "fullWidth": "MuiFormControl-fullWidth-4",
            "marginDense": "MuiFormControl-marginDense-3",
            "marginNormal": "MuiFormControl-marginNormal-2",
            "root": "MuiFormControl-root-1",
          }
        }
        component="div"
        disabled={false}
        error={false}
        fullWidth={true}
        margin="none"
        required={false}
        variant="standard"
      >
        <div
          className="MuiFormControl-root-1 MuiFormControl-fullWidth-4 label"
        >
          <WithStyles(WithFormControlContext(InputLabel))>
            <WithFormControlContext(InputLabel)
              classes={
                Object {
                  "animated": "MuiInputLabel-animated-13",
                  "disabled": "MuiInputLabel-disabled-7",
                  "error": "MuiInputLabel-error-8",
                  "filled": "MuiInputLabel-filled-14",
                  "focused": "MuiInputLabel-focused-6",
                  "formControl": "MuiInputLabel-formControl-10",
                  "marginDense": "MuiInputLabel-marginDense-11",
                  "outlined": "MuiInputLabel-outlined-15",
                  "required": "MuiInputLabel-required-9",
                  "root": "MuiInputLabel-root-5",
                  "shrink": "MuiInputLabel-shrink-12",
                }
              }
            >
              <InputLabel
                classes={
                  Object {
                    "animated": "MuiInputLabel-animated-13",
                    "disabled": "MuiInputLabel-disabled-7",
                    "error": "MuiInputLabel-error-8",
                    "filled": "MuiInputLabel-filled-14",
                    "focused": "MuiInputLabel-focused-6",
                    "formControl": "MuiInputLabel-formControl-10",
                    "marginDense": "MuiInputLabel-marginDense-11",
                    "outlined": "MuiInputLabel-outlined-15",
                    "required": "MuiInputLabel-required-9",
                    "root": "MuiInputLabel-root-5",
                    "shrink": "MuiInputLabel-shrink-12",
                  }
                }
                disableAnimation={false}
                muiFormControl={
                  Object {
                    "adornedStart": false,
                    "disabled": false,
                    "error": false,
                    "filled": false,
                    "focused": false,
                    "margin": "none",
                    "onBlur": [Function],
                    "onEmpty": [Function],
                    "onFilled": [Function],
                    "onFocus": [Function],
                    "required": false,
                    "variant": "standard",
                  }
                }
              >
                <WithStyles(WithFormControlContext(FormLabel))
                  className="MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
                  classes={
                    Object {
                      "disabled": "MuiInputLabel-disabled-7",
                      "error": "MuiInputLabel-error-8",
                      "focused": "MuiInputLabel-focused-6",
                      "required": "MuiInputLabel-required-9",
                    }
                  }
                  data-shrink={false}
                >
                  <WithFormControlContext(FormLabel)
                    className="MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
                    classes={
                      Object {
                        "asterisk": "MuiFormLabel-asterisk-22",
                        "disabled": "MuiFormLabel-disabled-18 MuiInputLabel-disabled-7",
                        "error": "MuiFormLabel-error-19 MuiInputLabel-error-8",
                        "filled": "MuiFormLabel-filled-20",
                        "focused": "MuiFormLabel-focused-17 MuiInputLabel-focused-6",
                        "required": "MuiFormLabel-required-21 MuiInputLabel-required-9",
                        "root": "MuiFormLabel-root-16",
                      }
                    }
                    data-shrink={false}
                  >
                    <FormLabel
                      className="MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
                      classes={
                        Object {
                          "asterisk": "MuiFormLabel-asterisk-22",
                          "disabled": "MuiFormLabel-disabled-18 MuiInputLabel-disabled-7",
                          "error": "MuiFormLabel-error-19 MuiInputLabel-error-8",
                          "filled": "MuiFormLabel-filled-20",
                          "focused": "MuiFormLabel-focused-17 MuiInputLabel-focused-6",
                          "required": "MuiFormLabel-required-21 MuiInputLabel-required-9",
                          "root": "MuiFormLabel-root-16",
                        }
                      }
                      component="label"
                      data-shrink={false}
                      muiFormControl={
                        Object {
                          "adornedStart": false,
                          "disabled": false,
                          "error": false,
                          "filled": false,
                          "focused": false,
                          "margin": "none",
                          "onBlur": [Function],
                          "onEmpty": [Function],
                          "onFilled": [Function],
                          "onFocus": [Function],
                          "required": false,
                          "variant": "standard",
                        }
                      }
                    >
                      <label
                        className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
                        data-shrink={false}
                      >
                        label
                      </label>
                    </FormLabel>
                  </WithFormControlContext(FormLabel)>
                </WithStyles(WithFormControlContext(FormLabel))>
              </InputLabel>
            </WithFormControlContext(InputLabel)>
          </WithStyles(WithFormControlContext(InputLabel))>
          <WithStyles(Input)
            fullWidth={true}
            name="password"
            onChange={[Function]}
            placeholder="hello"
            type="text"
          >
            <Input
              classes={
                Object {
                  "disabled": "MuiInput-disabled-26",
                  "error": "MuiInput-error-28",
                  "focused": "MuiInput-focused-25",
                  "formControl": "MuiInput-formControl-24",
                  "fullWidth": "MuiInput-fullWidth-30",
                  "input": "MuiInput-input-31",
                  "inputMarginDense": "MuiInput-inputMarginDense-32",
                  "inputMultiline": "MuiInput-inputMultiline-33",
                  "inputType": "MuiInput-inputType-34",
                  "inputTypeSearch": "MuiInput-inputTypeSearch-35",
                  "multiline": "MuiInput-multiline-29",
                  "root": "MuiInput-root-23",
                  "underline": "MuiInput-underline-27",
                }
              }
              fullWidth={true}
              name="password"
              onChange={[Function]}
              placeholder="hello"
              type="text"
            >
              <WithStyles(WithFormControlContext(InputBase))
                classes={
                  Object {
                    "disabled": "MuiInput-disabled-26",
                    "error": "MuiInput-error-28",
                    "focused": "MuiInput-focused-25",
                    "formControl": "MuiInput-formControl-24",
                    "fullWidth": "MuiInput-fullWidth-30",
                    "input": "MuiInput-input-31",
                    "inputMarginDense": "MuiInput-inputMarginDense-32",
                    "inputMultiline": "MuiInput-inputMultiline-33",
                    "inputType": "MuiInput-inputType-34",
                    "inputTypeSearch": "MuiInput-inputTypeSearch-35",
                    "multiline": "MuiInput-multiline-29",
                    "root": "MuiInput-root-23 MuiInput-underline-27",
                    "underline": null,
                  }
                }
                fullWidth={true}
                inputComponent="input"
                multiline={false}
                name="password"
                onChange={[Function]}
                placeholder="hello"
                type="text"
              >
                <WithFormControlContext(InputBase)
                  classes={
                    Object {
                      "adornedEnd": "MuiInputBase-adornedEnd-41",
                      "adornedStart": "MuiInputBase-adornedStart-40",
                      "disabled": "MuiInputBase-disabled-39 MuiInput-disabled-26",
                      "error": "MuiInputBase-error-42 MuiInput-error-28",
                      "focused": "MuiInputBase-focused-38 MuiInput-focused-25",
                      "formControl": "MuiInputBase-formControl-37 MuiInput-formControl-24",
                      "fullWidth": "MuiInputBase-fullWidth-45 MuiInput-fullWidth-30",
                      "input": "MuiInputBase-input-46 MuiInput-input-31",
                      "inputAdornedEnd": "MuiInputBase-inputAdornedEnd-52",
                      "inputAdornedStart": "MuiInputBase-inputAdornedStart-51",
                      "inputMarginDense": "MuiInputBase-inputMarginDense-47 MuiInput-inputMarginDense-32",
                      "inputMultiline": "MuiInputBase-inputMultiline-48 MuiInput-inputMultiline-33",
                      "inputType": "MuiInputBase-inputType-49 MuiInput-inputType-34",
                      "inputTypeSearch": "MuiInputBase-inputTypeSearch-50 MuiInput-inputTypeSearch-35",
                      "marginDense": "MuiInputBase-marginDense-43",
                      "multiline": "MuiInputBase-multiline-44 MuiInput-multiline-29",
                      "root": "MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27",
                    }
                  }
                  fullWidth={true}
                  inputComponent="input"
                  multiline={false}
                  name="password"
                  onChange={[Function]}
                  placeholder="hello"
                  type="text"
                >
                  <InputBase
                    classes={
                      Object {
                        "adornedEnd": "MuiInputBase-adornedEnd-41",
                        "adornedStart": "MuiInputBase-adornedStart-40",
                        "disabled": "MuiInputBase-disabled-39 MuiInput-disabled-26",
                        "error": "MuiInputBase-error-42 MuiInput-error-28",
                        "focused": "MuiInputBase-focused-38 MuiInput-focused-25",
                        "formControl": "MuiInputBase-formControl-37 MuiInput-formControl-24",
                        "fullWidth": "MuiInputBase-fullWidth-45 MuiInput-fullWidth-30",
                        "input": "MuiInputBase-input-46 MuiInput-input-31",
                        "inputAdornedEnd": "MuiInputBase-inputAdornedEnd-52",
                        "inputAdornedStart": "MuiInputBase-inputAdornedStart-51",
                        "inputMarginDense": "MuiInputBase-inputMarginDense-47 MuiInput-inputMarginDense-32",
                        "inputMultiline": "MuiInputBase-inputMultiline-48 MuiInput-inputMultiline-33",
                        "inputType": "MuiInputBase-inputType-49 MuiInput-inputType-34",
                        "inputTypeSearch": "MuiInputBase-inputTypeSearch-50 MuiInput-inputTypeSearch-35",
                        "marginDense": "MuiInputBase-marginDense-43",
                        "multiline": "MuiInputBase-multiline-44 MuiInput-multiline-29",
                        "root": "MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27",
                      }
                    }
                    fullWidth={true}
                    inputComponent="input"
                    muiFormControl={
                      Object {
                        "adornedStart": false,
                        "disabled": false,
                        "error": false,
                        "filled": false,
                        "focused": false,
                        "margin": "none",
                        "onBlur": [Function],
                        "onEmpty": [Function],
                        "onFilled": [Function],
                        "onFocus": [Function],
                        "required": false,
                        "variant": "standard",
                      }
                    }
                    multiline={false}
                    name="password"
                    onChange={[Function]}
                    placeholder="hello"
                    type="text"
                  >
                    <div
                      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
                      onClick={[Function]}
                    >
                      <input
                        aria-invalid={false}
                        className="MuiInputBase-input-46 MuiInput-input-31"
                        disabled={false}
                        name="password"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="hello"
                        required={false}
                        type="text"
                      />
                    </div>
                  </InputBase>
                </WithFormControlContext(InputBase)>
              </WithStyles(WithFormControlContext(InputBase))>
            </Input>
          </WithStyles(Input)>
        </div>
      </FormControl>
    </WithStyles(FormControl)>
  </TextField>
</Input>
`;
